src/MediawikiApi.php

```diff
@@ -351,14 +351,36 @@ private function getUserAgent() {
 	 * @param $result
 	 */
 	private function logWarnings( $result ) {
-		if ( is_array( $result ) && array_key_exists( 'warnings', $result ) ) {
-			foreach ( $result['warnings'] as $module => $warningData ) {
-				// Accomodate both formatversion=2 and old-style API results
-				$logPrefix = $module . ': ';
-				if ( isset( $warningData['*'] ) ) {
-					$this->logger->warning( $logPrefix . $warningData['*'], [ 'data' => $warningData ] );
-				} else {
-					$this->logger->warning( $logPrefix . $warningData['warnings'], [ 'data' => $warningData ] );
+		if ( is_array( $result ) ) {
+			// Let's see if there is 'warnings' key on the first level of the array...
+			if ( array_key_exists( 'warnings', $result ) ) {
+				foreach ( $result['warnings'] as $module => $warningData ) {
+					// Accomodate both formatversion=2 and old-style API results
+					$logPrefix = $module . ': ';
+					if ( isset( $warningData['*'] ) ) {
+						$this->logger->warning( $logPrefix . $warningData['*'], [ 'data' => $warningData ] );
+					} else {
+						$this->logger->warning( $logPrefix . $warningData['warnings'], [ 'data' => $warningData ] );
+					}
+
+					return;
+				}
+			}
+
+			// ...if no then go one level deeper and check there for them.
+			foreach ( $result as $value ) {
+				if ( !array_key_exists( 'warnings', $value ) ) {
```

Comment:
`$value` can be a string (see [test results](https://travis-ci.org/addwiki/mediawiki-api-base/jobs/318353872#L999)).